### DISCUSSION TITLE
brew install gcc: The post-install step did not complete successfully

### DISCUSSION THREAD

**Discussion by cobrexus** [Homebrew/discussions#3030] [Linux] [answered]
I installed Homebrew on Ubuntu on WSL. I have Windows 11 and I am learning Laravel, which requires WSL to run. I installed Homebrew because the instructor used it to install PHP, MySQL, Redis and other Laravel dependencies.

When I installed Homebrew with the command on the homepage and followed the next steps, it advised me to install GCC with `brew install gcc`, and I did, but I got this warning:

```
Warning: The post-install step did not complete successfully
You can try again using:
  brew postinstall gcc
```

When I run `brew postinstall gcc` I get this warning again.

`brew reinstall gcc` gives me the same result.

When I run `brew update` it says it's already up-to-date.

When I run `brew doctor`, it says that GCC is not installed:

```
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  isl@0.18

Warning: No developer tools installed.
Install Clang or run `brew install gcc`.
```

**Comment by SMillerDev** [comment]
Try running `brew postinstall --debug gcc` and post the output here.

**Comment by carlocab** [comment, marked as answer]
See https://docs.brew.sh/Homebrew-on-Linux#requirements.

**Reply by cobrexus** [reply]
Thanks, it works now

**Reply by deepakdp11** [reply]
please elaborate what worked

**Reply by isikyus** [reply]
I had the same issue as \@abzr1 describes.

It turned out I'd had errors from a previous `apt-get install` step, something like this:

```
Err:1 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 libc-dev-bin amd64 2.35-0ubuntu3.1
  404  Not Found [IP: 91.189.91.83 80]
```

Running `sudo apt-get update`, then the command `sudo apt-get install build-essential procps curl file git` from the linked document, fixed it for me.

**Reply by lack-coder** [reply]
how can i correct this Warning: The post-install step did not complete successfully
You can try again using:
  brew postinstall gcc

**Reply by isikyus** [reply]
> how can i correct this Warning: The post-install step did not complete successfully You can try again using: brew postinstall gcc

There are a number of commands that might help, in abzr1's original post, carlocab's link, and my comment above yours — have you tried any of those? If so, which ones? What was the result?

**Reply by RodrigoRVSN** [reply]
Had the same problem and doing the step-by-step commented by isikyus solved my issue here in WSL

**Comment by sudonam** [comment]
run `sudo apt install build-essential` cause brew need it

**Comment by LorenzoBettini** [comment]
Maybe I'm missing something, but by installing the requirements as listed here https://docs.brew.sh/Homebrew-on-Linux#requirements for Fedora, I already get gcc installed as a package, so it would not make sense to install it with Homebrew.

**Reply by carlocab** [reply]
It depends on which version of GCC  you want. If you don't care which version you're installing, then, yes, it probably doesn't make sense.

**Reply by LorenzoBettini** [reply]
Makes sense! ;)
thanks!
